baker/archival/ArchivalBaker.ts

```diff
@@ -0,0 +1,276 @@
+        [path.basename(dataFilename)]: `/${dataFilenameWithHash}`,
+        [path.basename(metadataFilename)]: `/${metadataFilenameWithHash}`,
+    }
+}
+
+const bakeOwidMjsFile = async (
+    srcPath: string,
+    destPath: string,
+    sourceMapFilename: string,
+    archiveDir: string
+) => {
+    const mjsContents = await fs.readFile(srcPath, "utf8")
+    const withSourceMap = mjsContents.replace(
+        /\/\/# sourceMappingURL=owid.mjs.map/g,
+        `//# sourceMappingURL=${sourceMapFilename}`
+    )
+
+    if (withSourceMap === mjsContents) {
+        console.error("Failed to replace sourceMappingURL in owid.mjs")
+    }
+    return await hashAndWriteFile(destPath, withSourceMap, archiveDir)
+}
+
+// we explicitly want owid.mjs.map to come first, so that we can replace the sourceMappingURL in owid.mjs
+// after we know the content-hashed filename of owid.map.mjs
+const ASSET_FILES = ["owid.mjs.map", "owid.mjs", "owid.css"]
+const IGNORED_FILES = [".vite"]
+export const bakeAssets = async (archiveDir: string) => {
+    const srcDir = path.join(projBaseDir, "dist/assets")
+    const targetDir = path.join(archiveDir, "assets")
+
+    await fs.mkdirp(targetDir)
+
+    const staticAssetMap: AssetMap = {}
+
+    const filesInDir = await fs.readdir(srcDir, { withFileTypes: true })
+
+    for await (const filename of ASSET_FILES) {
+        if (!filesInDir.some((dirent) => dirent.name === filename)) {
+            throw new Error(`Could not find ${filename} in ${srcDir}`)
+        }
+        const srcFile = path.join(srcDir, filename)
+
+        let outFilename: string
+        if (filename === "owid.mjs") {
+            const sourceMapFilename = path.basename(
+                staticAssetMap["owid.mjs.map"] ?? ""
+            )
+            if (!sourceMapFilename)
+                throw new Error("Could not find owid.mjs.map in staticAssetMap")
+
+            outFilename = await bakeOwidMjsFile(
+                srcFile,
+                path.join(targetDir, filename),
+                sourceMapFilename,
+                archiveDir
+            )
+        } else {
+            outFilename = await hashAndCopyFile(srcFile, targetDir, archiveDir)
+        }
+        staticAssetMap[filename] = `/${outFilename}`
+    }
+
+    const additionalFiles = new Set(
+        filesInDir.map((dirent) => dirent.name)
+    ).difference(new Set([...ASSET_FILES, ...IGNORED_FILES]))
+    if (additionalFiles.size > 0) {
+        console.warn(
+            `Found additional files in ${srcDir}:`,
+            Array.from(additionalFiles).join(", ")
+        )
+    }
+
+    return { staticAssetMap }
+}
+
+const archiveVariableIds = async (
+    variableIds: number[],
+    archiveDir: string
+) => {
+    const results = await pMap(
+        variableIds,
+        async (variableId) => {
+            return {
+                [variableId]: await bakeVariableDataFiles(
+                    variableId,
+                    archiveDir
+                ),
+            }
```

Comment:
There's a curiosity here in that we re-bake variable files even if they didn't change (their checksums are the same), but the grapher config of an underlying chart changed.

What's happening in practice is that the variable file is then re-fetched, re-hashed, and is then overriding the existing file under the same name.

It's not ideal, but in the grand scheme of things it's not a worry also - the median incremental update will probably touch less than 10 charts and less than 20 variables or so.